src/cpp/dds/domain/DomainParticipantImpl.cpp

```diff
@@ -0,0 +1,714 @@
+        return nullptr;
+    }
+
+    if(!att.remoteLocatorList.isValid())
+    {
+        logError(PARTICIPANT,"Remote Locator List for Publisher contains invalid Locator");
+        return nullptr;
+    }
+
+    if(!att.qos.checkQos() || !att.topic.checkQos())
+    {
+        return nullptr;
+    }
+
+    //TODO CONSTRUIR LA IMPLEMENTACION DENTRO DEL OBJETO DEL USUARIO.
+    PublisherImpl* pubimpl = new PublisherImpl(this, qos, att, listen);
+    Publisher* pub = new Publisher(pubimpl);
+    pubimpl->user_publisher_ = pub;
+    pubimpl->rtps_participant_ = rtps_participant_;
+
+    // Create InstanceHandle for the new publisher
+    GUID_t pub_guid = guid();
+    do
+    {
+        pub_guid.entityId = fastrtps::rtps::c_EntityId_Unknown;
+        rtps_participant_->get_new_entity_id(pub_guid.entityId);
+    } while (exists_entity_id(pub_guid.entityId));
+
+    InstanceHandle_t pub_handle(pub_guid);
+    pubimpl->handle_ = pub_handle;
+
+    //SAVE THE PUBLISHER INTO MAPS
+    std::lock_guard<std::mutex> lock(mtx_pubs_);
+    publishers_by_handle_[pub_handle] = pub;
+    publishers_[pub] = pubimpl;
+
+    return pub;
+}
+
+/* TODO
+Subscriber* DomainParticipantImpl::get_builtin_subscriber()
+{
+    logError(PARTICIPANT, "Not implemented.");
+    return nullptr;
+}
+*/
+
+/* TODO
+bool DomainParticipantImpl::ignore_participant(
+        const fastrtps::rtps::InstanceHandle_t& handle)
+{
+    (void)handle;
+    logError(PARTICIPANT, "Not implemented.");
+    return false;
+}
+*/
+
+/* TODO
+bool DomainParticipantImpl::ignore_topic(
+        const fastrtps::rtps::InstanceHandle_t& handle)
+{
+    (void)handle;
+    logError(PARTICIPANT, "Not implemented.");
+    return false;
+}
+*/
+
+/* TODO
+bool DomainParticipantImpl::ignore_publication(
+        const fastrtps::rtps::InstanceHandle_t& handle)
+{
+    (void)handle;
+    logError(PARTICIPANT, "Not implemented.");
+    return false;
+}
+*/
+
+/* TODO
+bool DomainParticipantImpl::ignore_subscription(
+        const fastrtps::rtps::InstanceHandle_t& handle)
+{
+    (void)handle;
+    logError(PARTICIPANT, "Not implemented.");
+    return false;
+}
+*/
+
+uint8_t DomainParticipantImpl::get_domain_id() const
+{
+    return static_cast<uint8_t>(att_.rtps.builtin.domainId);
+}
+
+/* TODO
+bool DomainParticipantImpl::delete_contained_entities()
+{
+    logError(PARTICIPANT, "Not implemented.");
+    return false;
+}
+*/
+
+bool DomainParticipantImpl::assert_liveliness()
+{
+    if (rtps_participant_->wlp() != nullptr)
+    {
+        return rtps_participant_->wlp()->assert_liveliness_manual_by_participant();
+    }
+    else
+    {
+        logError(PARTICIPANT, "Invalid WLP, cannot assert liveliness of participant");
+    }
+    return false;
+}
+
+bool DomainParticipantImpl::set_default_publisher_qos(
+        const fastdds::dds::PublisherQos& qos)
+{
+    if (&qos == &PUBLISHER_QOS_DEFAULT)
+    {
+        default_pub_qos_.set_qos(PUBLISHER_QOS_DEFAULT, true);
```

Comment:
In this case the function must return true.
```suggestion
        default_pub_qos_.set_qos(PUBLISHER_QOS_DEFAULT, true);
        return true;
```